### ISSUE TITLE
New environment options are missing selected status with screen readers

### ISSUE THREAD

**Issue by BrailleBennett** [modrinth/code#4285]
### Please confirm the following.

- [x] I checked the [existing issues](https://github.com/modrinth/code/issues?q=is%3Aissue) for duplicate problems
- [x] I have tried resolving the issue using the [support portal](https://support.modrinth.com)

### What browsers are you seeing the problem on?

Safari

### Describe the bug

When going to the new environment page on a project, all of the options for client side, single player, server side, etc. do not have the status of selected when they are selected according to a screen reader like VoiceOver or NVDA, this makes it functionally impossible to update or confidently verify my projects on the new system until this bug is resolved.

### Steps to reproduce

1. Enable any modern screen reader (E.G. VoiceOver, NVDA, Jaws)
2. Go to the settings page of a Modrinth project
3. Click on "Environment, new"
4. Observe that when clicking any options on this page the state of the button is not updated to reflect your change.

### Expected behavior

The site would correctly attribute the selected elements to allow screen readers to add their own styling to the buttons and speak what the screen reader user has configured to happen on selected elements.

**Comment by Prospector** [comment]
Just pushed a fix to this, let me know if there's any further changes that should be made. I gave them the radio button role, as well as created radio groups for their parents. I gave the sub-option groups titles, I hope that helps because I was concerned the suboptions would come across as confusing.
